core/drivers/versal_nvm.c

```diff
@@ -0,0 +1,944 @@
+
+	memcpy(buf, lbuf, sizeof(*buf));
+
+	return TEE_SUCCESS;
+}
+
+TEE_Result versal_efuse_read_sec_misc1(struct versal_efuse_sec_misc1_bits *buf)
+{
+	uint8_t lbuf[1024]__aligned_efuse = { 0 };
+	struct versal_nvm_read_req req = {
+		.efuse_id = EFUSE_READ_SEC_MISC1,
+	};
+
+	req.ibuf[0].buf = lbuf;
+	req.ibuf[0].len = sizeof(lbuf);
+
+	if (versal_nvm_read(&req))
+		return TEE_ERROR_GENERIC;
+
+	memcpy(buf, lbuf, sizeof(*buf));
+
+	return TEE_SUCCESS;
+}
+
+TEE_Result
+versal_efuse_read_boot_env_ctrl(struct versal_efuse_boot_env_ctrl_bits *buf)
+{
+	uint8_t lbuf[1024]__aligned_efuse = { 0 };
+	struct versal_nvm_read_req req = {
+		.efuse_id = EFUSE_READ_BOOT_ENV_CTRL,
+	};
+
+	req.ibuf[0].buf = lbuf;
+	req.ibuf[0].len = sizeof(lbuf);
+
+	if (versal_nvm_read(&req))
+		return TEE_ERROR_GENERIC;
+
+	memcpy(buf, lbuf, sizeof(*buf));
+
+	return TEE_SUCCESS;
+}
+
+TEE_Result versal_efuse_read_offchip_revoke_id(uint32_t *buf, size_t len,
+					       enum versal_nvm_offchip_id id)
+{
+	uint8_t lbuf[1024]__aligned_efuse = { 0 };
+	struct versal_nvm_read_req req = {
+		.efuse_id = EFUSE_READ_OFFCHIP_REVOCATION_ID,
+		.offchip_id = id, };
+
+	if (len < EFUSE_OFFCHIP_REVOCATION_ID_LEN)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	req.ibuf[0].buf = lbuf;
+	req.ibuf[0].len = sizeof(lbuf);
+
+	if (versal_nvm_read(&req))
+		return TEE_ERROR_GENERIC;
+
+	memcpy(buf, lbuf, EFUSE_REVOCATION_ID_LEN);
+
+	return TEE_SUCCESS;
+}
+
+TEE_Result versal_efuse_read_dec_only(uint32_t *buf, size_t len)
+{
+	uint8_t lbuf[1024]__aligned_efuse = { 0 };
+	struct versal_nvm_read_req req = {
+		.efuse_id = EFUSE_READ_DEC_EFUSE_ONLY,
+	};
+
+	if (len < EFUSE_DEC_ONLY_LEN)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	req.ibuf[0].buf = lbuf;
+	req.ibuf[0].len = sizeof(lbuf);
+
+	if (versal_nvm_read(&req))
+		return TEE_ERROR_GENERIC;
+
+	memcpy(buf, lbuf, EFUSE_DEC_ONLY_LEN);
+
+	return TEE_SUCCESS;
+}
+
+TEE_Result versal_efuse_read_puf_sec_ctrl(struct versal_efuse_puf_sec_ctrl_bits
+					  *buf)
+{
+	uint8_t lbuf[1024]__aligned_efuse = { 0 };
+	struct versal_nvm_read_req req = {
+		.efuse_id = EFUSE_READ_PUF_SEC_CTRL,
+	};
+
+	req.ibuf[0].buf = lbuf;
+	req.ibuf[0].len = sizeof(lbuf);
+
+	if (versal_nvm_read(&req))
+		return TEE_ERROR_GENERIC;
+
+	memcpy(buf, lbuf, sizeof(*buf));
+
+	return TEE_SUCCESS;
+}
+
+TEE_Result versal_efuse_read_puf(struct versal_efuse_puf_header *buf)
+{
+	uint8_t lbuf[1024]__aligned_efuse = { 0 };
+	struct versal_nvm_read_req req = {
+		.efuse_id = EFUSE_READ_PUF,
+	};
+
+	memcpy(lbuf, buf, sizeof(*buf));
+
+	req.ibuf[0].buf = lbuf;
+	req.ibuf[0].len = sizeof(lbuf);
+
+	if (versal_nvm_read(&req))
+		return TEE_ERROR_GENERIC;
+
+	memcpy(buf, lbuf, sizeof(*buf));
+
+	return TEE_SUCCESS;
+}
+
+/*
+ *  This functionality requires building the PLM with XNVM_ACCESS_PUF_USER_DATA
+ *  Calls will fail otherwise.
+ *  When available, efuse_read_puf becomes unavailable.
+ */
+TEE_Result versal_efuse_read_puf_as_user_fuse(struct versal_efuse_puf_user_fuse
```

Comment:
Please break the line as:
``` C
TEE_Result 
versal_efuse_read_puf_as_user_fuse(struct versal_efuse_puf_user_fuse *p)
```

Same for all the other functions where you're breaking the line between the type and parameter name.